### PR TITLE
Code Updates for using libvlc (2.1.1), fixes #33

### LINKED ISSUE #33: Integration with libvlc 2.1.1
@megabitdragon created a [new vlc2 branch](https://github.com/amahi/android/tree/vlc2) in this tree that tries to use libvlc 2.1.1.

He **only** added the library dependency without any other changes to the code. The application **will not build** as is in that branch.

The task in this issue is to update the code as necessary to account for the new libvlc SDK and make it 1) compile properly and 2) run properly.

You can check [this commit](https://goo.gl/BbyKmz) to get started by looking at what the VLC team did with their sample embedded player to get it adapted to the new version of the library. Presumably, our code may have to change in similar ways to adapt to the library.

Side note: no policy changes as to when invoke this player should be done. That is left for #17.

### PR DESCRIPTION
**Pull request by csoni111**
This pr is in reference to issue #33 on github and issue [#2014](https://bugs.amahi.org/issues/2014) on amahi issue tracker. 

There have been a lot of changes in the libvlc library since it was last used in this project. I have tried to make all the necessary changes required to make it work.

**The following changes have been made:**

- Moved maven url for libvlc from `buildscript repositories` to `dependencies repositories` in build.gradle
- New method for initializing LibVlc with some args added to it
- New MediaPlayer class added
- Replaced `setSurfaceSize` method previously given in `libvlc.IVideoPlayer` and implemented in `VideoService` **by** `onNewVideoLayout` given in `IVLCVout.OnNewVideoLayoutListener` and implemented in `ServerFileVideoActivity`
- Added new code implementation for scaling the surface view size according to the video's dimensions
- Removed Event Listeners from `VideoService` and implemented in `ServerFileVideoActivity` as provided in new `MediaPlayer.EventListener` interface
- Removed BusEvents and related classes, as they were no longer required

### CODE REVIEW

**Comment by megabitdragon** [comment]
I've tried this. The app builds fine but it crashes when trying to play movies. [Here](https://goo.gl/FS8Zf8) is the error for reference.

**Comment by csoni111** [comment]
Please Make a **Clean Build** and the error will be resolved.

You are getting the error because previously in `ServerFileVideoActivity` class there was an injected instance for `serverClient` field but in the updated code, it has been removed. The `Dagger` library used in the project makes separate classes for all the injected dependencies on building the project. You might be having previously built classes like this one `ServerFileVideoActivity$$InjectAdapter.java` in the build folder. Cleaning the build folder will resolve the error.

File: src/main/java/org/amahi/anywhere/activity/ServerFileVideoActivity.java [hunk 1]
```diff
@@ -24,59 +24,75 @@
...
-public class ServerFileVideoActivity extends Activity implements ServiceConnection,
-	SurfaceHolder.Callback,
+public class ServerFileVideoActivity extends Activity implements
+	ServiceConnection,
 	Runnable,
 	MediaController.MediaPlayerControl,
-	View.OnSystemUiVisibilityChangeListener
-{
-	@Inject
-	ServerClient serverClient;
```
Comment (csoni111) [inline src/main/java/org/amahi/anywhere/activity/ServerFileVideoActivity.java]: Just for reference: here ^ is the instance injection that was removed. @megabitdragon

**Comment by megabitdragon** [comment]
The clean build did the job. There are however some issues with it. Some might be vlc related and I'll take a look at those but there are others as well (e.g.: the controls don't show up when you tap screen, movie restarts when changing orientation). Good job so far.

**Comment by csoni111** [comment]
I will try to resolve the errors

**Comment by csoni111** [comment]
@megabitdragon I have tried to resolve some of the issues that you mentioned:

- Orientation change issue has been resolved.
- For controls not showing up after fullscreen, I am facing an odd bug: 
If I remove this toast message from [line#155](https://github.com/amahi/android/pull/48/commits/ea5e330d4d0c0dea3728604b3edfc937149bc599#diff-9584e161a48b19f7a905a0b167332822R155) in my [last commit]([url](https://github.com/amahi/android/pull/48/commits/ea5e330d4d0c0dea3728604b3edfc937149bc599)), the controls dont show up but if this toast is displayed then controls work as expected. I am not sure why this is happening.

I have another solution for this, in my another pr #71 I have added a `FullScreenHelper.java` class to make any activity go full screen and display controls on screen touch event. You may test it in that pr in the ImageViewer Activity. 
So after that gets merged we can rebase and use that class for making this video activity go fullscreen.

**Comment by csoni111** [comment]
Looks like `toggling video controls` is a long standing bug. I just found it here [#1683](https://bugs.amahi.org/issues/1683) referenced two years ago.

**Comment by megabitdragon** [comment]
Good work on this. The movie restarting when changing orientation is fixed. I see what you mean about the toast and video controls. I will take a look at it as well. Meanwhile I merged PR #71 and rebased the vlc2 branch. This might create some issues with your commits since the references were changed.

**Comment by csoni111** [comment]
No issues I have rebased my vlc2 branch accordingly and now I can use that `FullScreenHelper` class to resolve that `video controls issue`. Hopefully now it will be resolved!

**Comment by csoni111** [comment]
@megabitdragon the `video controls toggling issue` has now been fixed in my last commit. So till now following bugs have been fixed as per amahi bug tracker: [#2014](https://bugs.amahi.org/issues/2014), [#1683](https://bugs.amahi.org/issues/1683), [#1556](https://bugs.amahi.org/issues/1556), [#1416](https://bugs.amahi.org/issues/1416).

**Comment by csoni111** [comment]
Bug [#1446](https://bugs.amahi.org/issues/1446): `MediaController Leaks Window After Video Stop` and some sync issue between toolbar and media controls have now been resolved.

**Comment by megabitdragon** [comment]
I tried it it works as expected. One small issue I noticed is that when I change orientation in the emulator, the video only shows up in a corner. It works fine on device.
device-2017-03-13-195923

**Comment by cpg** [comment]
Hi guys, i decided to try this and after a full cleanup, it works quite well on all file formats I tried it on. Great job!

Should we merge it into master?

**Pull request merged**